src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -554,6 +555,18 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
 
                             continue;
 
+                        case "sourcelink":
+                            value = RemoveQuotesAndSlashes(value);
+                            if (string.IsNullOrEmpty(value))
+                            {
+                                AddDiagnostic(diagnostics, ErrorCode.ERR_NoFileSpec, arg);
+                            }
+                            else
+                            {
+                                sourceLink = ParseGenericPathToFile(value, diagnostics, baseDirectory);
```

Comment:
How do we expect those sourcelink files to be generated?

In my opinion, reporting errors at load-time in the debugger has a few problems:

1. The person building is not the same person that is debugging,
2. When you're debugging it is too late to fix (long iteration cycle),
3. I'm not sure what would be the error reporting experience when the debugger encounters a bad sourcelink file.

Those problems are worse if the sourcelink file is manually written, but I think they are still applicable if it's generated by a build task.

On the flip-side, I recognize that validation at compile-time may be cheap but it would not be foolproof (what if the URLs are invalid?). 
Also, it could couple the debugger and compiler.
Is that what you're worried about? Maybe that could be solved by generating a warning instead of an error.